src/main/java/com/yahoo/bullet/operations/aggregations/Distribution.java

```diff
@@ -0,0 +1,200 @@
+    public static final Error REQUIRES_POINTS_ERROR =
+            makeError("The DISTRIBUTION type requires at least one point",
+                      "Please add a list of numeric points, OR specify a number of equidistant points to generate" +
+                      "OR specify a start, end and increment (start < end, increment > 0) to generate points");
+    public static final Error REQUIRES_POINTS_PROPER_RANGE =
+            makeError(DistributionType.QUANTILE.getName() + " requires points in the proper range",
+                    "Please add or generate points: 0 <= point <= 1");
+    public static final Error REQUIRES_ONE_FIELD_ERROR =
+            makeError("The aggregation type requires exactly one field", "Please add exactly one field to fields");
+
+    /**
+     * Constructor that requires an {@link Aggregation}.
+     *
+     * @param aggregation An {@link Aggregation} with valid fields and attributes for this aggregation type.
+     */
+    @SuppressWarnings("unchecked")
+    public Distribution(Aggregation aggregation) {
+        super(aggregation);
+        entries = ((Number) config.getOrDefault(BulletConfig.DISTRIBUTION_AGGREGATION_SKETCH_ENTRIES,
+                                                DEFAULT_ENTRIES)).intValue();
+        int pointLimit = ((Number) config.getOrDefault(BulletConfig.DISTRIBUTION_AGGREGATION_MAX_POINTS,
+                                                       DEFAULT_MAX_POINTS)).intValue();
+        maxPoints = Math.min(pointLimit, aggregation.getSize());
+        this.aggregation = aggregation;
+
+        // The sketch is initialized in initialize!
+    }
+
+    @Override
+    public List<Error> initialize() {
+        Map<String, String> fields = aggregation.getFields();
+        if (Utilities.isEmpty(fields) || fields.size() != 1) {
+            return singletonList(REQUIRES_ONE_FIELD_ERROR);
+        }
+
+        Map<String, Object> attributes = aggregation.getAttributes();
+        if (Utilities.isEmpty(attributes)) {
+            return singletonList(REQUIRES_TYPE_ERROR);
+        }
+
+        String typeString = Utilities.getCasted(attributes, TYPE, String.class);
+        DistributionType type = SUPPORTED_DISTRIBUTION_TYPES.get(typeString);
+        if (type == null) {
+            return singletonList(REQUIRES_TYPE_ERROR);
+        }
+
+        // Initialize field since we have exactly 1
+        field = fields.keySet().stream().findFirst().get();
+
+        // Try to initialize sketch now
+        sketch = getSketch(entries, maxPoints, type, attributes);
+
+        if (sketch == null) {
+            return type == DistributionType.QUANTILE ? asList(REQUIRES_POINTS_ERROR, REQUIRES_POINTS_PROPER_RANGE) :
+                                                       singletonList(REQUIRES_POINTS_ERROR);
+        }
+        return null;
+    }
+
+    @Override
+    public void consume(BulletRecord data) {
+        Number value = Specification.extractFieldAsNumber(field, data);
```

Comment:
Yeah, it  should be in BulletRecord but this was the best place to put it without making a Utilities in parsing. Since the field String using the '.' notation is a parsing thing, I don't want to put it in a top-level Utilities.